client/src/store/authStore.ts

```diff
@@ -0,0 +1,57 @@
+import { create } from "zustand";
+
+interface AuthState {
+  token: string | null;
+  register: (name: string, email: string, password: string) => Promise<void>;
+  login: (email: string, password: string) => Promise<void>;
+  logout: () => void;
+}
+
+export const useAuthStore = create<AuthState>((set) => ({
+  token: null,
+
+  register: async (name, email, password) => {
+    const response = await fetch(`${import.meta.env.VITE_API_URL}${"/api/users/register"}`, {
```

Comment:
here you use redundant ${} for the string, it is more difficult to read.

you can write : 
` const response = await fetch(`${import.meta.env.VITE_API_URL}/api/users/login`, {})`

when you use ${} here always pass a param/code in javascript like ${variable}